### PR TITLE
chore(discover): Adding more codes for clickhouse errors

### PR DESCRIPTION
**Pull request by wmak**
- Right now everything is defaulting to QueryExecutionError which is
  getting hard for some more common errors
- Codes selected here are based on the most common exceptions

### CODE REVIEW

File: src/sentry/utils/snuba.py [hunk 1]
```diff
@@ -176,10 +176,41 @@ class QueryTooManySimultaneous(QueryExecutionError):
     """
 
 
+class QuerySizeExceeded(QueryExecutionError):
+    """
+    The generated query has excedeed the maximum length allowed by clickhouse
```
Comment (dashed) [inline src/sentry/utils/snuba.py:181]:
```suggestion
    The generated query has exceeded the maximum length allowed by clickhouse
```

**Review by dashed — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Review by Zylphrex — APPROVED** [review]
Note to self: handle these errors in data export as well

File: src/sentry/utils/snuba.py [hunk 2]
```diff
@@ -176,10 +176,41 @@ class QueryTooManySimultaneous(QueryExecutionError):
...
+class QueryConnectionFailed(QueryExecutionError):
+    """
+    The connection to clickhouse has failed, and so the query cannot be run
+    """
+
+
 clickhouse_error_codes_map = {
     43: QueryIllegalTypeOfArgument,
-    241: QueryMemoryLimitExceeded,
+    62: QuerySizeExceeded,
```
Comment (fpacifici) [inline src/sentry/utils/snuba.py:208]: Why do you need to have a special case for this? It is an internal error we should prevent on our own.

File: src/sentry/utils/snuba.py [hunk 3]
```diff
@@ -176,10 +176,41 @@ class QueryTooManySimultaneous(QueryExecutionError):
...
+    160: QueryExecutionTimeMaximum,
     202: QueryTooManySimultaneous,
+    241: QueryMemoryLimitExceeded,
+    271: DatasetSelectionError,
```
Comment (fpacifici) [inline src/sentry/utils/snuba.py:212]: Same, what is the plan for this? If we can fail queries this way we  should connvert them to HTTP400 inn Snuba.

**Pull request merged**